### ISSUE TITLE
display.flip()  Gives Window surface is invalid error (1698)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#935]
#  **Issue &numero;1698 opened by *[MossabDiae](https://github.com/MossabDiae)* at 2020-05-02 18:19:10**

**WARNING** : I'm totally new to pygame and relatively new to python , I did my research on the problem but I'm stuck.
with that said, the following code will generate a surface, while waiting for QUIT event keeps refreshing the screen using `pygame.display.flip()`
```
import sys
import pygame
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")
    
    # the game loop/ to update surfaces AND listen to events
    while True:
        # when ever an event (eg key press) is caused , the loop will trigger
        # all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()
        
        # then update screen
        pygame.display.flip()

run_game()
```
the following error will result 
```
pygame 2.0.0.dev6 (SDL 2.0.10, python 3.7.6)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "make_window.py", line 19, in <module>
    run_game()
  File "make_window.py", line 17, in run_game
    pygame.display.flip()
pygame.error: Window surface is invalid, please call SDL_GetWindowSurface() to get a new surface
```

I confirm that **the code works fine in** `pygame 1.9.6` but due to high CPU usage I had to update .

**More informations** 
> Distro: MX-18.1_x64
> Wm: i3-gaps<hr>

#  Comments

# # #  **[MightyJosip](https://github.com/MightyJosip)* commented at 2020-05-02 19:31:19*

Hi, thanks for your report. Can you tell us do you do anything before the game crashes, or it crash the moment you run it.
<hr>

# # #  **[MossabDiae](https://github.com/MossabDiae)* commented at 2020-05-02 19:53:33*

> Hi, thanks for your report. Can you tell us do you do anything before the game crashes, or it crash the moment you run it.

Hi, thanks for your reply .
I do nothing , it crushes directly the moment I run it .
<hr>

# # #  **[MightyJosip](https://github.com/MightyJosip)* commented at 2020-05-02 20:02:16*

There were a few changes recently to the flip function that could resolve this problem. We will soon release pygame 2.0.0.dev8. I will mention you here once we do it and tell you how to install it (it will maybe fix your problem). One last question from me. Do you get the same error with the following code:

```
import sys
import pygame
import time


def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")

    time.sleep(1)
    #  the game loop/ to update surfaces AND listen to events
    while True:
        #  when ever an event (eg key press) is caused , the loop will trigger
        #  all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()

        #  then update screen
        pygame.display.flip()


run_game()
```

If that fails for now you can maybe use this code

```
import sys
import pygame

def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")

    #  the game loop/ to update surfaces AND listen to events
    while True:
        #  when ever an event (eg key press) is caused , the loop will trigger
        #  all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()

        #  then update screen
        pygame.display.update()

run_game()
```
<hr>

# # #  **[MossabDiae](https://github.com/MossabDiae)* commented at 2020-05-02 20:18:29*

Ok thank you .
As for the first code : the test **failed** (same traceback)
the second code **works fine** (high cpu usage but maybe because of the loop)

**TL;DR** 
for people having same issue (Window surface is invalid, please call SDL_GetWindowSurface)
1- install `pygame 2.0.0.dev8` : `pip install pygame==2.0.0.dev8`
OR
2- replace `display.flip()`  with `pygame.display.update()`
OR
3- downgrade to `pygame 1.9.6` and use `pygame.mixer.quit()` to fix high cpu usage problem
<hr>

# # #  **[MightyJosip](https://github.com/MightyJosip)* commented at 2020-05-02 20:30:25*

To reduce cpu usage use clock class
<hr>

# # #  **[MossabDiae](https://github.com/MossabDiae)* commented at 2020-05-02 20:37:55*

> To reduce cpu usage use clock class

I'm sorry I don't want to take much of your time , but if it's straightforward can you make the proper change for that in my code .
<hr>

# # #  **[MightyJosip](https://github.com/MightyJosip)* commented at 2020-05-02 20:45:26*

```
import sys
import pygame

clock = pygame.time.Clock()
FPS = 60
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")

    #  the game loop/ to update surfaces AND listen to events
    while True:
        #  when ever an event (eg key press) is caused , the loop will trigger
        #  all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()

        #  then update screen
        pygame.display.update()
        clock.tick(FPS)

run_game()
```
If you would like you can also join pygame discord server for help (https://discord.gg/r8yreB6).
<hr>

# # #  **[MossabDiae](https://github.com/MossabDiae)* commented at 2020-05-02 20:53:41*

Really appreciate it , THANK YOU !
I'll make sure to register in discord and join pygame server if I have any future problems .
Salam
<hr>

# # #  **[MightyJosip](https://github.com/MightyJosip)* commented at 2020-05-05 22:00:42*

@mossaybo Can you install the newest pygame version with `pip install pygame==2.0.0.dev8` and then run the code with `pygame.display.flip()` statement. If yes can you say do you still get the same error.
<hr>

# # #  **[MossabDiae](https://github.com/MossabDiae)* commented at 2020-05-10 20:00:23*

> @mossaybo Can you install the newest pygame version with `pip install pygame==2.0.0.dev8` and then run the code with `pygame.display.flip()` statement. If yes can you say do you still get the same error.

sure, (sorry for the delay)
so I tried my first code and I confirm **it's working fine** 

pygame 2.0.0.dev8 (SDL 2.0.12, python 3.7.6)
problem **SOLVED**
thank you.

```
import sys
import pygame
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")
    
    # the game loop/ to update surfaces AND listen to events
    while True:
        # when ever an event (eg key press) is caused , the loop will trigger
        # all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()
        
        # then update screen
        pygame.display.flip()

run_game()
```
